dm/dm-best-practices.md

```diff
@@ -0,0 +1,234 @@
+
+- Clustered indexes + `AUTO_RANDOM`
+
+    This solution can retain the benefits of using clustered indexes and avoid the write hotspot problem. It requires less effort for customization. You can modify the schema attribute when you switch to use TiDB as the write database. In subsequent queries, if you have to use the ID column to sort data, you can use the [`AUTO_RANDOM`](/auto-random.md) ID column and left shift 5 bits to ensure the order of the query data. For example:
+
+    ```sql
+    CREATE TABLE t (a bigint PRIMARY KEY AUTO_RANDOM, b varchar(255));
+    Select  a, a<<5 ,b from t order by a <<5 desc
+    ```
+
+The following table summarizes the pros and cons of each solution.
+
+| Scenario | Recommended solution | Pros | Cons |
+| :--- | :--- | :--- | :--- |
+| <li>TiDB will act as the primary and write-intensive database. </li><li>The business logic strongly depends on the continuity of the primary key IDs.</li> | Create the table with non-clustered indexes and set `SHARD_ROW_ID_BIT`. Use `SEQUENCE` as the primary key column.  | It can avoid data write hotspots and ensure the continuity and monotonic increment of business data. | <li>The throughput capacity of data write is decreased to ensure data write continuity. </li><li>The performance of primary key queries is decreased.</li> |
+| <li>TiDB will act as the primary and write-intensive database. </li><li>The business logic strongly depends on the increment of the primary key IDs.</li>  | Create the table with non-clustered indexes and set `SHARD_ROW_ID_BIT`. Use an application ID generator to generate the primary key IDs. | It can avoid data write hotspots, guarantee the performance of data write, guarantee the increment of business data, but cannot guarantee continuity. | <li>You need to customize the application. </li><li>External ID generators strongly depend on the clock accuracy and might introduce failure risks.</li> |
+| <li>TiDB will act as the primary and write-intensive database. </li><li>The business logic does not depend on the continuity of the primary key IDs.</li> | Create the table with clustered indexes and set `AUTO_RANDOM` for the primary key column. | <li>It can avoid data write hotspots and has excellent query performance of the primary keys. </li><li>You can smoothly switch from `AUTO_INCREMENT` to `AUTO_RANDOM`.</li> | <li>The primary key IDs are random. </li><li>The write throughput ability is limited. </li><li>It is recommended to sort the business data by using the insert time column. </li><li>If you have to use the primary key ID to sort data, you can left shift 5 bits to query, which can guarantee the increment of the data.</li> |
+| TiDB will act as the read-only database. | Create the table with non-clustered indexes and set `SHARD_ROW_ID_BIT`. Keep the primary key column consistent with the data source. | <li>It can avoid data write hotspots. </li><li>It requires less customization cost. </li>| The query performance of the primary keys is impacted. |
```

Comment:
```suggestion
| TiDB will act as a read-only database. | Create tables with non-clustered indexes and set `SHARD_ROW_ID_BIT`. Keep the primary key column consistent with the data source. | <li>It can avoid data write hotspots. </li><li>It requires less customization cost. </li>| The query performance of primary keys is impacted. |
```